### PR TITLE
Thermal diffusivity and Peclet number evaluation capability

### LINKED ISSUE #20476: Create Peclet number aux kernel for NS module
## Reason
The Peclet number defines the ratio between advective and conductive heat transport. It's important to know for determining whether to, for instance, upwind energy advection.

## Design
Twin of #20359/#20361

## Impact
Users will be able to visualize elemental Peclet numbers and make decisions about how to handle energy advection.

### PR DESCRIPTION
**Pull request by lindsayad**
- Thermal diffusivity is computed via a functor material (`ThermalDiffusivityFunctorMaterial`)
- The Peclet number is computed via aux kernel (`PecletNumberFunctorAux`)

Closes #20476

### CODE REVIEW

File: modules/navier_stokes/doc/content/source/auxkernels/PecletNumberFunctorAux.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+!syntax description /AuxKernels/PecletNumberFunctorAux
+
+## Overview
+
+The `PecletNumberFunctorAux` object computes the element/cell Peclet number
+given by
+
+\begin{equation}
+Re = \frac{h u}{\alpha}
```
Comment (GiudGiud) [inline modules/navier_stokes/doc/content/source/auxkernels/PecletNumberFunctorAux.md:11]:
```suggestion
Pe = \frac{h u}{\alpha}
```

File: modules/navier_stokes/src/auxkernels/PecletNumberFunctorAux.C [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+#include "MooseMesh.h"
+#include "NS.h"
+
+registerMooseObject("NavierStokesApp", PecletNumberFunctorAux);
+
+InputParameters
+PecletNumberFunctorAux::validParams()
+{
+  InputParameters params = AuxKernel::validParams();
+  params.addClassDescription("Computes u*L/alpha.");
```
Comment (GiudGiud) [inline modules/navier_stokes/src/auxkernels/PecletNumberFunctorAux.C:20]:
```suggestion
  params.addClassDescription("Computes the Peclet number: u*L/alpha.");
```

File: modules/navier_stokes/test/tests/auxkernels/peclet-number-functor-aux/fv-thermal.i [hunk 3]
```diff
@@ -0,0 +1,196 @@
+mu = 1
+rho = 1
+k = 1
+cp = 1
+
+[GlobalParams]
+  velocity_interp_method = 'rc'
+  advected_interp_method = 'average'
```
Comment (GiudGiud) [inline modules/navier_stokes/test/tests/auxkernels/peclet-number-functor-aux/fv-thermal.i:8]: maybe add a note that bcs of the Peclet number, this scheme is OK ?

Comment (lindsayad) [reply modules/navier_stokes/test/tests/auxkernels/peclet-number-functor-aux/fv-thermal.i:8]:
```suggestion
  # Maximum cell Peclet number is ~.1 so energy transport is stable without upwinding
  advected_interp_method = 'average'
```

**Review by GiudGiud — APPROVED** [review]

**Pull request merged**